clients/venice-admin-tool/src/main/java/com/linkedin/venice/ZkCopier.java

```diff
@@ -0,0 +1,196 @@
+package com.linkedin.venice;
+
+import static com.linkedin.venice.VeniceZkPaths.ADMIN_TOPIC_METADATA;
+import static com.linkedin.venice.VeniceZkPaths.EXECUTION_IDS;
+import static com.linkedin.venice.VeniceZkPaths.PARENT_OFFLINE_PUSHES;
+import static com.linkedin.venice.VeniceZkPaths.ROUTERS;
+import static com.linkedin.venice.VeniceZkPaths.STORES;
+import static com.linkedin.venice.VeniceZkPaths.STORE_GRAVEYARD;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FileUtils;
+
+
+public class ZkCopier {
+  private static final Set<String> VENICE_ZK_PATHS = new HashSet<>(
+      Arrays.asList(ADMIN_TOPIC_METADATA, EXECUTION_IDS, PARENT_OFFLINE_PUSHES, ROUTERS, STORE_GRAVEYARD, STORES));
+
+  /**
+   * Extract Venice-specific paths from an input text file to an output text file.
+   * @param inputPath absolute path to the ZK snapshot input text file, which contains one path per line
+   * @param outputPath absolute path to output text file with extracted Venice-specific paths, which contains one path per line
+   * @param clusterNames set of cluster names
+   * @param basePath base path for ZK
+   */
+  public static void extractVenicePaths(
+      String inputPath,
+      String outputPath,
+      Set<String> clusterNames,
+      String basePath) {
+    try {
+      // read ZK snapshot input file and store all paths in a list
+      File inputFile = new File(inputPath);
+      List<String> zkPaths = FileUtils.readLines(inputFile);
+
+      // write Venice-specific paths to output file
+      ArrayList<String> venicePaths = getVenicePaths(new ArrayList<>(zkPaths), clusterNames, basePath, VENICE_ZK_PATHS);
+      File outputFile = new File(outputPath);
+      FileUtils.writeLines(outputFile, venicePaths);
+    } catch (IOException e) {
+      throw new VeniceException(e.getMessage());
+    }
+  }
+
+  /**
+   * Get Venice-specific paths filtered by base path, cluster names, and required paths from a list of ZK paths.
+   */
+  public static ArrayList<String> getVenicePaths(
+      ArrayList<String> zkPaths,
+      Set<String> clusterNames,
+      String basePath,
+      Set<String> requiredPaths) {
+    TreeNode requiredPathsTree = buildRequiredPathsTree(clusterNames, basePath, requiredPaths);
+    TreeNode zkPathsTree = listToTree(zkPaths, basePath);
+    getVenicePathsHelper(zkPathsTree, requiredPathsTree);
+    return treeToList(zkPathsTree);
+  }
+
+  private static void getVenicePathsHelper(TreeNode zkPathsTree, TreeNode requiredPathsTree) {
```

Comment:
Let's add a javadoc and mention here that this will modify the `zkPathsTree` directly